oak_functions/loader/src/server.rs

```diff
@@ -468,72 +456,30 @@ where
             let response = task.now_or_never();
 
             match response {
-                None => policy.create_error_response("response not available"),
+                None => policy.create_http_response(FunctionsResponse {
+                    status: FunctionsStatusCode::PolicyViolation as i32,
+                    body: "Reason: response not available".as_bytes().to_vec(),
+                }),
                 Some(response) => match response {
-                    Err(err) => policy.create_error_response(format!("{}", err).as_str()),
+                    // Error when getting the response from the tokio task
+                    Err(err) => policy.create_http_response(FunctionsResponse {
+                        status: FunctionsStatusCode::InternalServerError as i32,
+                        body: format!("{}", err).as_bytes().to_vec(),
+                    }),
                     Ok(response) => match response {
-                        Err(err) => policy.create_error_response(format!("{}", err).as_str()),
-                        Ok(rsp) => {
-                            // All responses from Oak-Functions have an additional
-                            // `oak-policy-padding` header.
-                            let size = estimate_size(&rsp)?;
-                            match size.cmp(&(policy.constant_response_size_bytes as i64)) {
-                                Ordering::Greater => {
-                                    policy.create_error_response("response too large")
-                                }
-                                Ordering::Less => policy.pad_response(rsp),
-                                Ordering::Equal => Response::<Body>::try_from(rsp)
-                                    .context("couldn't get the response"),
-                            }
-                        }
+                        // Error generated by `function`
+                        Err(err) => policy.create_http_response(FunctionsResponse {
+                            status: FunctionsStatusCode::InternalServerError as i32,
+                            body: format!("{}", err).as_bytes().to_vec(),
+                        }),
+                        Ok(rsp) => policy.create_http_response(rsp),
```

Comment:
Why is `create_http_response` applied within each branch of this match? It seems it should just applied once around it.

Reply:
Fixed.